### PR TITLE
[Testing] Update tests to use attributes for data providers

### LINKED ISSUE #20202: make test case classes compatible with PHPUnit 10+
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#58182
| PR author(s) | @xabbuh
| Merged in    | 7.2

We created this issue to not forget to document this new feature. We would really appreciate if you can help us with this task. If you are not sure how to do it, please ask us and we will help you.

To fix this issue, please create a PR against the 7.2 branch in the [symfony-docs repository](https://github.com/symfony/symfony-docs).

Thank you! :smiley:

### PR DESCRIPTION
**Pull request by javiereguiluz**
Fixes #20202.

### CODE REVIEW

File: best_practices.rst [hunk 1]
```diff
@@ -433,6 +432,11 @@ checks that all application URLs load successfully::
         }
     }
 
+.. note::
+
+    The ``#[DataProvider]`` attribute is available in PHPUnit 10 and later versions.
+    In previous versions, use the ``@dataPRovider`` PHPdoc annotation.
```
Comment (alexisLefebvre) [inline best_practices.rst]:
```suggestion
    In previous versions, use the ``@dataProvider`` PHPdoc annotation.
```

Typo with `r`.

File: best_practices.rst [hunk 2]
```diff
@@ -433,6 +432,11 @@ checks that all application URLs load successfully::
         }
     }
 
+.. note::
+
+    The ``#[DataProvider]`` attribute is available in PHPUnit 10 and later versions.
+    In previous versions, use the ``@dataProvider`` PHPdoc annotation.
```
Comment (MrYamous) [inline best_practices.rst]:
```suggestion
    In previous versions, use the ``@dataProvider`` PHPDoc annotation.
```
To make DOCtor-RST happy :)

Comment (javiereguiluz) [reply best_practices.rst]: Good catch! Fixed. Thanks.

**Comment by javiereguiluz** [comment]
Let's ask @symfony/docs folks if they agree on using PHPUnit 10+ in 7.2 branch ... or if they prefer to wait ... or even if they prefer to use it in earlier branches. Thanks.

**Comment by wouterj** [comment]
I think our focus should be to be consistent with the PHPunit version across the documentation.

And the main thing is then deprecation handling (e.g. https://symfony.com/doc/current/setup/upgrade_major.html#deprecations-in-phpunit). This means we should at least switch to PHPunit 11.3 and update articles like that one to use PHPunit's [error handling features](https://docs.phpunit.de/en/11.3/error-handling.html#limiting-issues-to-your-code). If we do this, I think it's OK to keep using PHPunit 9 (with a big warning admonition) for the PhpUnitBridge documentation until we support it (probably in 7.3 or 7.4).

For me, it's either that or keep using PHPunit 9 until we have support for PHPunit 11.3+ in the PhpUnitBridge. I don't see much value in only updating this attribute.

**Comment by javiereguiluz** [comment]
OK, let's close this without merging and let's update PHPUnit to 11 in Symfony 7.3 or 7.4/8.0. Thanks!

**Pull request closed without merging**